src/steps/mobile_devices/index.ts

```diff
@@ -20,25 +23,43 @@ export async function fetchMobileDevices(
 
   const accountEntity = await getAccountEntity(context);
 
-  await client.iterateMobileDevices(async (device) => {
-    const deviceEntity = createMobileDeviceEntity(device);
+  try {
+    await client.iterateMobileDevices(async (device) => {
+      const deviceEntity = createMobileDeviceEntity(device);
+
+      if (await jobState.hasKey(deviceEntity._key)) {
+        logger.info(
+          { _key: deviceEntity._key },
+          'Duplicate device entity _key found',
+        );
+        return;
+      }
 
-    if (await jobState.hasKey(deviceEntity._key)) {
-      logger.info(
-        { _key: deviceEntity._key },
-        'Duplicate device entity _key found',
+      await context.jobState.addEntity(deviceEntity);
+      await context.jobState.addRelationship(
+        createAccountManagesMobileDeviceRelationship({
+          accountEntity,
+          deviceEntity,
+        }),
       );
+    });
+  } catch (err) {
+    if (err instanceof IntegrationProviderAuthorizationError) {
+      context.logger.info(
+        { err },
+        'Could not ingest mobile device information.',
+      );
+      context.logger.publishEvent({
+        name: 'missing_scope',
+        description: `Could not ingest mobile device data. Missing required scope(s) (scopes=${client.requiredScopes.join(
+          ', ',
+        )}).  Additionally, Admin Email provided in configuration must have the Admin API Privliege "Manage Devices and Settings" enabled.`,
```

Comment:
Typo + small grammar suggestion:

```suggestion
        )}).  Additionally, the admin email provided in configuration must have the Admin API privilege "Manage Devices and Settings" enabled.`,
```